### PR TITLE
Implement GenericIteratorObject

### PR DESCRIPTION
**Pull request by ksh8281**
Signed-off-by: Seonghyun Kim <sh8281.kim@samsung.com>

### CODE REVIEW

**Review by bbrto21 — APPROVED** [review]
LGTM!

File: test/cctest/testapi.cpp [hunk 1]
```diff
@@ -309,6 +309,58 @@ class ShellPlatform : public PlatformRef {
...
+{
+    PersistentRefHolder<ContextRef> context = ContextRef::create(instance);
+
+    Evaluator::execute(context, [](ExecutionStateRef* state) -> ValueRef* {
+        ContextRef* context = state->context();
+
+        {
+            FunctionObjectRef::NativeFunctionInfo nativeFunctionInfo(AtomicStringRef::create(context, "print"), builtinPrint, 1, true, false);
+            FunctionObjectRef* buildFunctionObjectRef = FunctionObjectRef::create(state, nativeFunctionInfo);
+            context->globalObject()->defineDataProperty(state, StringRef::createFromASCII("print"), buildFunctionObjectRef, true, true, true);
```
Comment (clover2123) [inline test/cctest/testapi.cpp:349]: It seems that native function `builtinPrint` is not used anywhere.

**Comment by clover2123** [comment]
One more thing, (just suggestion) how about adding `@@toStringTag` property for GenericIteratorObject?
Other iterator objects have own property for this.

File: test/cctest/testapi.cpp (hunk 1 above)
Comment (ksh8281) [reply test/cctest/testapi.cpp:349]: You are right. the print function is just needed for writing test case conveniently.
If you want I can remove the function.

**Comment by ksh8281** [comment]
I added the @@toStringTag with "Iterator" you can find it from GlobalObject.cpp:949

**Review by clover2123 — APPROVED** [review]
Oh I skipped that @@toStringTag part.
LGTM

**Pull request merged**